mpmath/matrices/matrices.py

```diff
@@ -280,38 +280,39 @@ def __init__(self, *args, **kwargs):
         # multiple times, when calculating the inverse and when calculating the
         # determinant
         self._LU = None
+
         if "force_type" in kwargs:
             warnings.warn("The force_type argument was removed, it did not work"
                 " properly anyway. If you want to force floating-point or"
                 " interval computations, use the respective methods from `fp`"
                 " or `mp` instead, e.g., `fp.matrix()` or `iv.matrix()`."
                 " If you want to truncate values to integer, use .apply(int) instead.")
-        if isinstance(args[0], (list, tuple)):
-            if isinstance(args[0][0], (list, tuple)):
```

Comment:
Are there reasons we have such huge changes in the constructor just for one failing case (``matrix([])``)?  I would expect something like
```diff
diff --git a/mpmath/matrices/matrices.py b/mpmath/matrices/matrices.py
index 7447ce1..f1b49b3 100644
--- a/mpmath/matrices/matrices.py
+++ b/mpmath/matrices/matrices.py
@@ -287,7 +287,11 @@ def __init__(self, *args, **kwargs):
                 " or `mp` instead, e.g., `fp.matrix()` or `iv.matrix()`."
                 " If you want to truncate values to integer, use .apply(int) instead.")
         if isinstance(args[0], (list, tuple)):
-            if isinstance(args[0][0], (list, tuple)):
+            if not args[0]:
+                A = args[0]
+                self.__rows = 0
+                self.__cols = 0
+            elif isinstance(args[0][0], (list, tuple)):
                 # interpret nested list as matrix
                 A = args[0]
                 self.__rows = len(A)
```